pkg/common/auth_credentials/auth_credentials.go

```diff
@@ -1,34 +1,52 @@
 package auth_credentials
 
 import (
+	"context"
 	"fmt"
 	"regexp"
 	"strings"
 
 	envoyServiceAuthV3 "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"
+	v1 "k8s.io/api/core/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type AuthCredentials interface {
 	GetCredentialsFromReq(*envoyServiceAuthV3.AttributeContext_HttpRequest) (string, error)
+	GetCredentialsFromCluster(context.Context, map[string]string) ([]string, error)
```

Comment:
I think we're mixing two different things here, one being the data structure that tells how request credentials are expected to be supplied and a tool to extract such value from a given instance of request, and another thing which is a service to fetch an array of strings from somewhere.

Reply:
One could see it as a module in charge of managing credentials, since those strings are not any simple string. Anyway, where would you see this making more sense?

Reply:
For now, getting secrets from the cluster is an APIKey auth identity config-specific thing. It could be part of `APIKey`. Otherwise inside yet another separate module that is only about getting resources from the cluster.

I think in the end practically everything that happens in "identity" could be seen as about managing/handling credentials. Soon, those will be: openid, api_key, oauth, openshift, hmac, etc. IMO, better not opening this door of accommodating in this AuthCredentials module anything that has to do with credentials because that can end up being a lot. In fact, I think AuthCredentials should actually be about "CredentialsConfig", i.e. the struct to store that part of the config that holds where the credentials are carried in the request.